lib/matplotlib/axes/_base.py

```diff
@@ -2336,15 +2336,23 @@ def _update_line_limits(self, line):
                                       updatex=updatex, updatey=updatey)
         self.ignore_existing_data_limits = False
 
-    def add_patch(self, p):
+    def add_patch(self, p, relim=True):
```

Comment:
This is needed because adding the extend triangles would cause a relim, despite the fact that they are not to be used to set the limits of the axes, and indeed they are outside the axes.   We could maybe change this to simply not relim if the patch is not in data co-ordinates...